apps/fluent-tester/src/TestComponents/Notification/NotificationTest.tsx

```diff
@@ -9,156 +14,262 @@ const svgProps: SvgIconProps = {
+      />
+      <TextInput value={message} onChangeText={setMessage} style={styles.textInput} />
+      <View style={styles.notification}>
+        <Notification
+          variant={variant}
+          onPress={() => {
+            console.log('Notification tapped');
+          }}
+        >
+          {message}
+        </Notification>
+      </View>
+    </View>
   );
 };
 
-const PrimaryOutlineBarTest: React.FunctionComponent = () => {
+const PrimaryWithAutoHide: React.FunctionComponent = () => {
+  const [visible, setVisible] = React.useState(false);
+  const onButtonPress = () => setVisible(!visible);
+
+  const [hidden, setHidden] = React.useState<boolean>(!visible);
+  const height = React.useRef(new Animated.Value(0)).current;
+
+  const show = () => {
+    setHidden(false);
+    Animated.sequence([
+      Animated.timing(height, {
+        toValue: SHOW_HEIGHT,
+        duration: NOTIFICATION_SHOW_DURATION,
+        useNativeDriver: USE_NATIVE_DRIVER_IOS,
+        easing: Easing.elastic(NOTIFICATION_BOUNCINESS),
+      }),
+      Animated.delay(AUTO_HIDE_DURATION),
+      Animated.timing(height, {
+        toValue: 0,
+        duration: HIDE_DURATION,
+        useNativeDriver: USE_NATIVE_DRIVER_IOS,
+      }),
+    ]).start(() => {
+      setHidden(true);
+      setVisible(false);
+    });
+  };
+
+  const hide = () => {
+    Animated.timing(height, {
+      toValue: 0,
+      duration: HIDE_DURATION,
+      useNativeDriver: USE_NATIVE_DRIVER_IOS,
+    }).start(() => {
+      setHidden(true);
+    });
+  };
+
+  React.useLayoutEffect(() => {
+    if (visible) {
+      show();
+    } else {
+      hide();
+    }
+  }, [visible, height]);
+
+  const animatedViewProps = {
+    transform: [{ translateY: height }],
+  };
+
   return (
-    <Notification
-      variant={'primaryOutlineBar'}
-      onPress={() => {
-        console.log('Notification tapped');
-      }}
-    >
-      Mail Sent
-    </Notification>
+    <View>
+      <Button appearance="subtle" onClick={onButtonPress} style={styles.button}>
+        {visible ? 'Hide' : 'Show'}
+      </Button>
+      {!hidden && (
+        <Animated.View style={[animatedViewProps]}>
+          <Notification
+            variant={'primary'}
+            action="Undo"
+            onPress={() => {
+              console.log('Notification tapped');
+            }}
+            onActionPress={() => {
+              console.log('Undo tapped');
+            }}
+          >
+            Mail Archived
+          </Notification>
+        </Animated.View>
+      )}
+    </View>
   );
 };
+const PrimaryBarWithAutoHide: React.FunctionComponent = () => {
+  const [visible, setVisible] = React.useState(false);
+  const onButtonPress = () => setVisible(!visible);
+
+  const [hidden, setHidden] = React.useState<boolean>(!visible);
+  const height = React.useRef(new Animated.Value(0)).current;
+
+  const show = () => {
+    setHidden(false);
+    Animated.sequence([
```

Comment:
Did you happen to consider using LayoutAnimations for this animation?  Apologies for asking so late after you've been talking through this animation w/ Saad for a while, there are a couple things we need to mind with animations cross-platform.

react-native-win32 doesn't have a Native Animation Driver, so it will use the JS Animation Driver and worse busy-lock the app for the duration of the animation.  If we move forward w/ Animated here should the animations be disabled for win32, or the component not offered on win32 until the Native Animation Driver is available?

LayoutAnimation as an alternative does have an implementation for win32 and would change the implementation to only use layout to determine position vs laying out to a pre-animation position and then animating to its intended position.

Actually :( I'm realizing the more pressing issue for win32 is that the `transform` property is poorly implemented and hit testing won't work as expected (the button will activate if you click its laid out position, not its transformed position). I realize this is coming as a macOS/iOS-firstpriority component but the cross-plat impl will need to be kept off of win32 or not animated for win32 until we get the platform gaps filled that I mentioned above :\

Reply:
i would like to comment notification so far is very mobile concept. i haven't seen anything in desktop design.

Reply:
> i would like to comment notification so far is very mobile concept. i haven't seen anything in desktop design.

Would limiting platforms down to iOS/android make sense?

Reply:
I think this will be a mobile only FURN component in the long run. I saw web had a similar looking "Alert", but Ididn't see it in FluentUI React v9. I imagine if/when FURN implements that, it would be a separate component. 

Regarding win32 animation concerns, I guess this is another point towards "Let's keep the animation method separate so that the client can customize it". I asked Joanna to not touch `Notification.tsx` as much as possible and try to keep the Notification _view_ and Notification _animation_ separate as much as possible :)